api/nodes/serializers.py

```diff
@@ -139,11 +141,20 @@ def update(self, node, validated_data):
             node.add_tag(new_tag, auth=auth)
         for deleted_tag in (old_tags - current_tags):
             node.remove_tag(deleted_tag, auth=auth)
+
+        if 'is_public' in validated_data:
+            privacy_key = 'public' if validated_data.pop('is_public') else 'private'
+            try:
+                node.set_privacy(privacy_key, auth=auth, log=True, save=True)
```

Comment:
Note: When https://openscience.atlassian.net/browse/OSF-4668 is addressed, this should be refactored so that `Node.set_privacy` is called within `Node.update`.